[GLib] Allow default world to be used in WebView async JS execution API

q66 opened the pull request:
#### bd12d92e5f93b162f8646cfccd7ecf770dca2a4b
<pre>
[GLib] Allow default world to be used in WebView async JS execution API
<a href="https://bugs.webkit.org/show_bug.cgi?id=243224">https://bugs.webkit.org/show_bug.cgi?id=243224</a>

Reviewed by Michael Catanzaro and Carlos Garcia Campos.

The patch loosens the API to allow it to do what the documentation says,
as well as clarifies the documentation a little. Additionally, it fixes
tests to match the intended behavior.

* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkit_web_view_run_async_javascript_function_in_world):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebView.cpp:
(testWebViewRunAsyncFunctions):
(testWebViewRunJavaScript):

Canonical link: <a href="https://commits.webkit.org/252947@main">https://commits.webkit.org/252947@main</a>
</pre>

mcatanzaro reviewed (approved):
Thanks for fixing this and adding a new test.

Commit:
[GLib] Allow default world to be used in WebView async JS execution API
https://bugs.webkit.org/show_bug.cgi?id=243224

Reviewed by Michael Catanzaro and Carlos Garcia Campos.

The patch loosens the API to allow it to do what the documentation says,
as well as clarifies the documentation a little. Additionally, it fixes
tests to match the intended behavior.

* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkit_web_view_run_async_javascript_function_in_world):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebView.cpp:
(testWebViewRunAsyncFunctions):
(testWebViewRunJavaScript):

Canonical link: https://commits.webkit.org/252947@main
